plugins/woocommerce-admin/client/tasks/fills/shipping/index.js

```diff
@@ -328,12 +334,153 @@ export class Shipping extends Component {
+						'We recommend the following shipping options based on your location. You can manage your shipping options again at any time in WooCommerce Shipping settings.',
+						'woocommerce'
+					),
+					content: (
+						<ShippingRates
+							buttonText={ __(
+								'Save shipping options',
+								'woocommerce'
+							) }
+							shippingZones={ this.state.shippingZones }
+							onComplete={ () => {
+								const { id } = task;
+								optimisticallyCompleteTask( id );
+								invalidateResolutionForStoreSelector();
+								this.completeStep();
+							} }
+							createNotice={ createNotice }
+						/>
+					),
+				},
+				label_printing: {
+					label: __(
+						'Enable shipping label printing and discounted rates',
+						'woocommerce'
+					),
+					description: __(
+						'Save time and fulfill your orders with WooCommerce Shipping. You can manage it at any time in WooCommerce Shipping Settings.',
+						'woocommerce'
+					),
+					content: (
+						<>
+							<WCSBanner />
+							<Plugins
+								onComplete={ ( plugins, response ) => {
+									createNoticesFromResponse( response );
+									recordEvent(
+										'tasklist_shipping_label_printing',
+										{
+											install: true,
+											plugins_to_activate:
+												pluginsToActivate,
+										}
+									);
+									this.completeStep();
+								} }
+								onError={ ( errors, response ) =>
+									createNoticesFromResponse( response )
+								}
+								onSkip={ () => {
+									recordEvent(
+										'tasklist_shipping_label_printing',
+										{
+											install: false,
+											plugins_to_activate:
+												pluginsToActivate,
+										}
+									);
+									getHistory().push(
+										getNewPath( {}, '/', {} )
+									);
+									onComplete();
+								} }
+								pluginSlugs={ pluginsToActivate }
+							/>
+							{ ! isJetpackConnected && (
+								<Text
+									variant="caption"
+									className="woocommerce-task__caption"
+									size="12"
+									lineHeight="16px"
+									style={ { display: 'block' } }
+								>
+									{ interpolateComponents( {
+										mixedString: agreementText,
+										components: {
+											link: (
+												<Link
+													href={
+														'https://wordpress.com/tos/'
+													}
+													target="_blank"
+													type="external"
+												>
+													<></>
+												</Link>
+											),
+										},
+									} ) }
+								</Text>
+							) }
+						</>
+					),
+				},
+				store_location: {
+					label: __( 'Set your store location', 'woocommerce' ),
+					description: __(
+						'Add your store location to help us calculate shipping rates and the best shipping options for you. You can manage your store location again at any time in WooCommerce Settings General.',
+						'woocommerce'
+					),
+					buttonText: __( 'Save store location', 'woocommerce' ),
+				},
+			};
+
+			for ( const i in shippingSmartDefaultsSteps ) {
```

Comment:
@moon0326 What do you think if we use array type for `shippingSmartDefaultsSteps` and reassign it to `steps` if `shippingSmartDefaultsEnabled`?  Then we don't need to use for loops here. We can assign connect step to a variable to reuse it to avoid duplicate code.

```js
const connectStep = {...};
let steps = [
   ...
];

if ( this.shippingSmartDefaultsEnabled  ) {
    steps = [ .... ];
}
```

Alternatively, I think we can improve this since shippingSmartDefaultsSteps is an object:

```js
let steps = [
 ....
];


steps.map( ( step ) =>
	shippingSmartDefaultsSteps.hasOwnProperty( step.key )
		? {
				...step,
				...shippingSmartDefaultsSteps[ step.key ],
			}
		: step
);
```

Reply:
Yup! That works better 👍 

My old school habit stepped in :) I'll update it to your suggestion before merging.